app/routes/task_routes.py

```diff
@@ -0,0 +1,135 @@
+from flask import Blueprint, jsonify, abort, make_response, request
+from app import db
+from app.models.task import Task
+from datetime import datetime
+import os
+import requests
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+# ================================
+# Create one task 
+# ================================
+@task_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    try:
+        new_task = Task.create(request_body)
+    except KeyError:
+        return make_response({"details": "Invalid data"}, 400)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify({"task": new_task.to_json()}), 201
+
+# ==================================
+# Get all tasks  
+# ==================================
+@task_bp.route("", methods=["GET"])
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query == 'asc':
+        all_tasks = Task.query.order_by(Task.title.asc())
+    elif sort_query == 'desc':
+        all_tasks = Task.query.order_by(Task.title.desc())
+    else:
+        all_tasks = Task.query.all()
+    
+    results_list = []
+    for task in all_tasks:
+        results_list.append(task.to_json())
+
+    return jsonify(results_list), 200
+
+# ==================================
+# Get one task by id number
+# ==================================
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    return jsonify({"task": validate_id(Task, task_id).to_json()}), 200
+
+# ==================================
+# Update one task 
+# ==================================
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    request_body = request.get_json()
+    task = validate_id(Task, task_id)
+    task.update(request_body)
+
+    db.session.commit()
+
+    return jsonify({"task": task.to_json()}), 200
+
+# ==================================
+# Delete one task by id
+# ==================================
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = validate_id(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response({"details":f'Task {task.task_id} "{task.title}" successfully deleted'}, 200)
+
+# ==================================
+# update one task's completeness
+# ==================================
+@task_bp.route("/<task_id>/<mark>", methods=["PATCH"])
+def mark_tasks_complete_or_incomplete(task_id, mark):
+    task = validate_id(Task, task_id)
+    task.completed_at = datetime.now() if mark == "mark_complete" else None
```

Comment:
Nice, I like how succinct this is, and saves having to define two different endpoints.

There's a downside in that I can provide other `mark` values than are acceptable, so `PATCH /tasks/13/mark_foo` would be the same as `PATCH /tasks/13/mark_incomplete` (or as a scarier example, `PATCH /tasks/13/mark_complet`, where I misspelled "complete").

But this is still a good idea, and there's a pretty easy fix.